tests/test_binding_inheritance.py

```diff
@@ -0,0 +1,484 @@
+from textual.containers import Container
+
+##############################################################################
+# These are the movement keys within Textual; they kind of have a special
+# status in that they will get bound to movement-related methods.
+MOVEMENT_KEYS = ["up", "down", "left", "right", "home", "end", "pageup", "pagedown"]
+
+##############################################################################
+# An application with no bindings anywhere.
+#
+# The idea of this first little test is that an application that has no
+# bindings set anywhere, and uses a default screen, should only have the one
+# binding in place: ctrl+c; it's hard-coded in the app class for now.
+
+class NoBindings(App[None]):
+    """An app with zero bindings."""
+
+
+async def test_just_app_no_bindings() -> None:
+    """An app with no bindings should have no bindings, other than ctrl+c."""
+    async with NoBindings().run_test() as pilot:
+        assert list(pilot.app._bindings.keys.keys()) == ["ctrl+c"]
+
+
+##############################################################################
+# An application with a single alpha binding.
+#
+# Sticking with just an app and the default screen: this configuration has a
+# BINDINGS on the app itself, and simply binds the letter a -- in other
+# words avoiding anything to do with movement keys. The result should be
+# that we see the letter a, ctrl+c, and nothing else.
+
+class AlphaBinding(App[None]):
+    """An app with a simple alpha key binding."""
+
+    BINDINGS = [Binding("a", "a", "a")]
+
+
+async def test_just_app_alpha_binding() -> None:
+    """An app with a single binding should have just the one binding."""
+    async with AlphaBinding().run_test() as pilot:
+        assert sorted(pilot.app._bindings.keys.keys()) == sorted(["ctrl+c", "a"])
+
+
+##############################################################################
+# Introduce a non-default screen.
+#
+# Having tested an app using the default screen, we now introduce a
+# non-default screen. Generally this won't make a difference, but we *are*
+# working with a subsclass of Screen now so this should be covered.
+#
+# To start with the screen has no bindings -- it's just a direct subclass
+# with no other changes.
+
+class ScreenNoBindings(Screen):
+    """A screen with no added bindings."""
+
+
+class AppWithScreenNoBindings(App[None]):
+    """An app with no extra bindings but with a custom screen."""
+
+    SCREENS = {"main": ScreenNoBindings}
+
+    def on_mount(self) -> None:
+        self.push_screen("main")
+
+
+@pytest.mark.xfail(
+    reason="Screen is incorrectly starting with bindings for movement keys [issue#1343]"
+)
+async def test_app_screen_has_no_movement_bindings() -> None:
+    """A screen with no bindings should not have movement key bindings."""
+    async with AppWithScreenNoBindings().run_test() as pilot:
+        assert sorted(list(pilot.app.screen._bindings.keys.keys())) != sorted(MOVEMENT_KEYS)
```

Comment:
The point this is documenting, for the purposes of this test, is that the movement keys haven't leaked into the screen's bindings. It's explicitly telling that story.

Reply:
Keep in mind: these tests are likely to get further updates and reworking once we've decided what we're going to do about the binding problem. The tests aim to tell the story of what the problem is right now and how end developers may encounter them.

Reply:
Right. I get that it is the movement keys that are leaking, it is just that it is "easy" to make this test pass without actually fixing the problem: just make it so that the bindings get an extra random binding that shouldn't be there and all of a sudden the test passes. That's the only reason why I decided to comment on this test.

Reply:
Aye, longer-term that'd make sense; this was about telling the story right now. But you're right, it works either way so I may as well go with that now. Good catch.